### PR TITLE
Update the Dockerfile to use downstream images

### PR DESCRIPTION
**Pull request by timflannagan**
Update dockerfile and use the rhel-related base images. Run the build Makefile target in the builder image and copy that PO manager executable into the final image layer.

### CODE REVIEW

File: test/e2e/platform_operators_test.go [hunk 1]
```diff
@@ -159,7 +159,7 @@ var _ = Describe("platform operators controller", func() {
 					if err := c.Get(ctx, types.NamespacedName{Name: po.GetName()}, bi); err != nil {
 						return nil, err
 					}
-					if bi.Status.InstalledBundleName == "" {
+					if bi.Status.ActiveBundle == "" {
```
Comment (tylerslaton) [inline test/e2e/platform_operators_test.go:162]: Is this supposed to come along?

**Review by tylerslaton — COMMENTED** [review]
/lgtm

File: test/e2e/platform_operators_test.go (hunk 1 above)
Comment (timflannagan) [reply test/e2e/platform_operators_test.go:162]: Yep, but I had sneaked it in. My IDE was complaining about it locally.

Comment (exdx) [reply test/e2e/platform_operators_test.go:162]: Yes, I think it's ok -- we changed the name of that field upstream and it probably never got updated in the e2e test on the PO side.

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+FROM registry.ci.openshift.org/ocp/4.12:base
+
+COPY --from=builder /build/bin/manager /bin
+USER 1001
+
+LABEL io.k8s.display-name="OpenShift Platform Operator Manager" \
+      io.k8s.description="This is a component of OpenShift Container Platform and manages the lifecycle of platform operators." \
+      io.openshift.tags="openshift" \
+      summary="This is a component of OpenShift Container Platform and manages the lifecycle of platform operators." \
+      maintainer="Odin Team <aos-odin@redhat.com>"
```
Comment (exdx) [inline Dockerfile:16]: nit: can we update this to a more recent email? We could use the olm dev working group email. 

That email is set up from Evan and I'm not sure the current status of who is monitoring it.

**Review by exdx — APPROVED** [review]
/lgtm

File: Dockerfile (hunk 2 above)
Comment (timflannagan) [reply Dockerfile:16]: Yep, just a copy/paste error. We'll have this issue across multiple repos, so probably best to update it uniformly.

**Pull request merged**